### ISSUE TITLE
Unneded iter for loop

### ISSUE THREAD

**Issue by Snowiiii** [rust-lang/rust-clippy#14430]
### Summary

Hello, While coding on my Rust project i encountered something that would definitely make sense to have as a clippy lint.

Problematic code
```rust
let array = [1, 2, 3];
for (i, _) in array.iter().enumerate() {
     // the array data is not used we only need the len 
}
```
Better alternative
```rust
let array = [1, 2, 3];
for i in 0..array.len() {
     // we only use the len
}
```
This is cleaner and also may faster since we don't have to iterate over the array anymore

### Lint Name

uneeded_iter_loop

### Reproducer

I tried this code:

```rust
let array = [1, 2, 3];
for (i, _) in array.iter().enumerate() {
     // the array data is not used we only need the len 
}
```

I expected to see this happen:
A clippy warn

Instead, this happened:
No clippy warn

### Version

```text
rustc: 1.85-stable
```
